data_processing/services/radiology_service.py

```diff
@@ -0,0 +1,136 @@
+        "window": fields.Boolean(description="Toggle window function", required=False, example=True),
+        "window.low_level": fields.Float(description="Low window value", required=False, example=-100),
+        "window.high_level": fields.Float(description="High window value", required=False, example=100),
+    }
+)
+
+# param models
+generate_scan_model = api.model("Generate Volumetric Image",
+    {
+        "job_tag": fields.String(description="Tag/name of output record", required=True, example='my_nrrd'),
+        "input_name": fields.String(description="Tag/name of input image record", required=True, example='dicoms'),
+        "file_ext": fields.String(description="A valid ITK image file extention", required=True, example='nrrd'),
+    }
+)
+
+@api.route('/mind/api/v1/window_dicom/<cohort_id>/<container_id>/submit', methods=['POST'])
+class API_window_dicom(Resource):
+    @api.expect(window_dicom_model, validate=True)
+    def post(self, cohort_id, container_id):
+        """Submit a scale and window CT dicom Job"""
+        job_id = str(uuid.uuid4())
+        future = executor.submit (window_dicom_with_container, cohort_id, container_id, request.json)
+        return f"Submitted job {job_id} with future {future}"
+
+@api.route('/mind/api/v1/extract_voxels/<cohort_id>/<container_id>/submit', methods=['POST'])
+class API_extract_voxels(Resource):
+    @api.expect(extract_voxels_model, validate=True)
+    def post(self, cohort_id, container_id):
+        """Submit an extract radiomics Job"""
+        job_id = str(uuid.uuid4())
+        return "Not implimented yet"
+        #future = executor.submit (extract_voxels_with_container, cohort_id, container_id, request.json)
+        #return f"Submitted job {job_id} with future {future}"
+
+@api.route('/mind/api/v1/extract_radiomics/<cohort_id>/<container_id>/submit', methods=['POST'])
+class API_extract_radiomics(Resource):
+    @api.expect(extract_radiomics_model, validate=True)
+    def post(self, cohort_id, container_id):
+        """Submit an extract radiomics Job"""
+        job_id = str(uuid.uuid4())
+        future = executor.submit (extract_radiomics_with_container, cohort_id, container_id, request.json)
+        return f"Submitted job {job_id} with future {future}"
+
+@api.route('/mind/api/v1/generate_scan/<cohort_id>/<container_id>/submit', methods=['POST'])
+class API_generate_scan(Resource):
+    @api.expect(generate_scan_model, validate=True)
+    def post(self, cohort_id, container_id):
+        """Submit a generate scan Job"""
+        job_id = str(uuid.uuid4())
+        future = executor.submit (generate_scan_with_container, cohort_id, container_id, request.json)
+        return f"Submitted job {job_id} with future {future}"
+
+
+@api.route('/service/health', methods=['GET'])
+class API_heatlh(Resource):
+    def get(self):
+        try:
+            executor.submit(print, "Test")
```

Comment:
Nice health check! do we need a pool submit here?

Reply:
Yes-- I encountered one case where the executor errors out and you can't submit jobs. This health check ensures the executor can accept submitted tasks.